<launch>
    <arg name="venv"    default="$(optenv ROS_GTTS_VENV)" />
    
    <arg name="ns"          default="tts" />
    <arg name="string_input_topic"  default="string_input" />
    <arg name="audio_data_output_topic" default="audio" />
    <arg name="launch_audio_play" default="true" />

    <arg name="language" default="en" />
    <!-- You can check the available languages and corresponding abbreviations by running 
    ```
    python -c "import gtts; [print(lang, '\t', name) for lang, name in gtts.lang.tts_langs().items()]"
    ```
    in your command line
    -->
    
    <!-- Remap the audio output topic -->
    <remap from="audio_data_output" to="/audio/$(arg audio_data_output_topic)" if="$(arg launch_audio_play)" />
    <remap from="audio_data_output" to="$(arg audio_data_output_topic)" unless="$(arg launch_audio_play)" />
    <!-- Launch the audio_play node if requested by the user -->
    <include file="$(find audio_play)/launch/play.launch" if="$(arg launch_audio_play)" >
        <arg name="audio_topic" value="$(arg audio_data_output_topic)" />
    </include>
    <!-- Launch the TTS node -->
    <node pkg="ros_gtts" type="gtts_node.py" name="gtts_node" launch-prefix="$(arg venv)" ns="$(arg ns)" output="screen" args="--lang $(arg language)" >
        <remap from="string_input" to="$(arg string_input_topic)" />
    </node>
</launch>